### ISSUE TITLE
Bugs with @font-face rules

### ISSUE THREAD

**Issue by autopaideia** [ben-eb/postcss-merge-rules#10]
I'm having some issues with this plugin and font-face rules. Here's my original issue:

```css
.one, .two, .three {
  font-family: "lorem";
  font-weight: normal;
}

.four {
  font-family: "lorem", serif;
  font-weight: normal;
}

.five {
  font-family: "lorem";
  font-weight: normal;
}

@font-face {
  font-family: "lorem";
  font-weight: normal;
  src: url(/assets/lorem.eot);
  src: url(/assets/lorem.eot?#iefix) format("embedded-opentype"),
       url(/assets/lorem.woff) format("woff"),
       url(/assets/lorem.ttf) format("truetype");
}

.six {
  width: auto;
}
```

This triggers an error originating from `canMerge` in postcss-merge-rules/dist/index.js when the font face rule tries to merge with the others. I thought the fix would just be adding a check to the top of that function:

```javascript
function canMerge (ruleA, ruleB) {
    // Can't merge if one of the rules has no selector
    if (!ruleA.selector || !ruleB.selector) {
        return false;
    }

    let a = list.comma(ruleA.selector);
    ...
```

...which does stop the plugin from throwing errors, but after some more digging I found a deeper issue that I think has something to do with the way font-face rules are being handled. For example:

```css
.foo {
  font-weight: normal;
}

.bar {
  font-family: "my-font";
  font-weight: normal;
}

@font-face {
  font-family: "my-font";
  font-weight: normal;
  src: url("my-font.ttf");
}
```

Will output:

```css
.foo{font-weight:400}@font-face{font-family:my-font;font-weight:400}@font-face{src:url(my-font.ttf)}
```

However, if you put the `@font-face` rule at the top in that example it will output correctly:

```css
@font-face{font-family:my-font;font-weight:400;src:url(my-font.ttf)}.bar,.foo{font-weight:400}.bar{font-family:my-font}
```

**Comment by ben-eb** [comment]
Thanks for the detailed report; a fix was published in `2.0.3` which is available on npm now. :smile:

**Comment by ben-eb** [comment]
No problem! :beers:
